### ISSUE TITLE
Interogate NCPA API and NCPA Crash.

### ISSUE THREAD

**Issue by metabsd** [NagiosEnterprises/ncpa#398]
Hi,

I already do that for a lot of server but 2 in particular crash each time I try to get all the definition from 

`GET /api?token=ourtokent HTTP/1.1" 301 582 0.000000`

The only thing I have in the log (In debug mode) is 
```
2017-10-26 14:40:06,486:DEBUG:handler:Initializing WebSocket
2017-10-26 14:40:06,486:DEBUG:handler:Validating WebSocket request
2017-10-26 14:40:06,486:INFO:handler:SOMEIP - - [2017-10-26 14:40:06] "GET /api?token=ourtokent HTTP/1.1" 301 582 0.000000
2017-10-26 14:40:06,532:DEBUG:handler:Initializing WebSocket
2017-10-26 14:40:06,532:DEBUG:handler:Validating WebSocket request
2017-10-26 14:40:07,671:DEBUG:psapi:Imported windowscounters into the API tree.
2017-10-26 14:40:07,671:DEBUG:psapi:Imported windowslogs into the API tree. 
```

NCPA is install on Windows 2008 R2 Standard. The default language setting is English.

Than you in advance for you help!

**Comment by jomann09** [comment]
Is this something that is happening a lot? Doesn't seem to be giving any errors out. Could it possibly be locking up the process or does it completely crash? I have some other stuff to test on 2008 R2 so I may try to replicate this when I test that too.

**Comment by metabsd** [comment]
We have to monitor those servers. I install another agent for the moment. Keep me in touch if you can replicate it. The service is running and the process is not dead but all the check crash and we not see any log in the log file.

**Comment by jomann09** [comment]
Was that the only log file that you found anything in? I will take a look a this asap.

**Comment by metabsd** [comment]
This is the only thing I see. I think Windows edition of NCPA is less verbose that Linux edition.

**Comment by jomann09** [comment]
So I installed it on my test 2008 R2 and it was working for me. Can you think of anything that might be different on those servers that may be causing it?

**Comment by metabsd** [comment]
French language is install.

**Comment by jomann09** [comment]
I'm going to set this to pending requirement. I am really going to try to do my best to get NCPA 3 out so that we can start using Python 3. I did some testing this morning and it seems like the unicode conversions are going to be a pain in Python 2, and after some testing it looks like the encoding is the reason that there are parts of the API that break with non-english languages in Windows.

**Comment by jomann09** [comment]
For the record I think that this is solved with 2.1, after doing some testing I noticed that there were some weird issues with encoding/decoding. Hopefully this will work in the new 2.1 version.

I'll leave this open so you can try this again in 2.1 and verify if you're still having issues.

**Comment by dTrimon** [comment]
Hello, just to tell That with version 2.1.3 I have the same behavior! Anyone having an idea of what can happen here?

**Comment by jomann09** [comment]
I didn't make any changes that would fix this in 2.1.3. I'm still not able to get it working properly in different languages (I don't know why python struggles with it so badly) but I think that python 3 will fix it, hopefully.

**Comment by jomann09** [comment]
I am pretty sure this was related to #460 so I am going to close it since it should be resolved in 2.1.5 and if not, please let me know.
